avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -332,32 +335,54 @@ public static String getTypeId(Schema schema) {
      */
     public static String getFullName(Schema schema) {
         switch (schema.getType()) {
-        case RECORD:
-        case ENUM:
-        case FIXED:
-            String namespace = schema.getNamespace();
-            String name = schema.getName();
-            if (namespace == null) {
-                return schema.getName();
-            }
-            if (namespace.endsWith("$")) {
-                return namespace + name;
-            }
-            StringBuilder sb = new StringBuilder(1 + namespace.length() + name.length());
-            // 28-Feb-2020: [dataformats-binary#195] somewhat complicated logic of trying
-            //     to support differences between avro-lib 1.8 and 1.9...
-            //     Check if this is a nested class
-            String nestedClassName = sb.append(namespace).append('$').append(name).toString();
-            try {
-                Class.forName(nestedClassName);
-                return nestedClassName;
-            } catch (ClassNotFoundException e) {
-                // Could not find a nested class, must be a regular class
-                sb.setLength(0);
-                return sb.append(namespace).append('.').append(name).toString();
-            }
+            case RECORD:
+            case ENUM:
+            case FIXED:
+                String namespace = schema.getNamespace();
+                String name = schema.getName();
+                String key = namespace + "." + name;
```

Comment:
`namespace` might be `null` here (not when a schema is generated from a java class, but if using hand-crafted or schemas generated from avdl files)

Reply:
Just double checked, `schema.getFullName()` should be suitable as a cache key here.

Reply:
Hi there @baharclerode! Thank you for additional comments -- and yes, I realize that Avro's choice of "interesting" ways of dealing with stuff sort of forces our hand here.
Would be happy to merge a PR for removing helper class as key, use `schema.getFullName()` if that should work?

One other related question: if we had a setting to select between "1.8-and-earlier" vs "1.9-and-later" modes, would it be possible to use different logic, avoiding Class lookup or at least alleviate it?
Or does the problem exist regardless of this, with 1.9 and above?
(I realize that right now lack of context makes it impossible to have such a setting, but I still hope we can make `AvroSchemaHelper` become non-static and/or take some context object we control which could pass configuration, and ideally contain per-mapper (or per-schema or whatever) cache).